api/v1/helpers/verbs/doGet.js

```diff
@@ -37,49 +37,57 @@ function doGet(req, res, next, props) {
   const fields = reqParams.fields ? reqParams.fields.value : null;
   const scopes = props.getScopes ? props.getScopes : [];
 
+  const opts = u.buildFieldList(reqParams, props.model);
+
   // only cache requests with no params
   if (props.cacheEnabled && !fields) {
     const cacheKey = reqParams.key.value;
 
     redisCache.get(cacheKey, (cacheErr, reply) => {
       if (cacheErr || !reply) {
         // if err or no reply, get from db and set redis cache
-        u.findByKey(props, req.swagger.params, scopes)
-        .then((o) => {
+        u.findByKey(props, reqParams, scopes)
+        .then((returnedObject) => {
           resultObj.dbTime = new Date() - resultObj.reqStartTime;
-          u.logAPI(req, resultObj, o);
-          res.status(httpStatus.OK).json(u.responsify(o, props, req.method));
+          u.logAPI(req, resultObj, returnedObject);
```

Comment:
Logging has to happen AFTER removing fields from response because we're logging response size.

Reply:
Done.